cmd/juju/caas/gke.go

```diff
@@ -0,0 +1,237 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package caas
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/cmd/juju/interact"
+)
+
+type gke struct {
+	CommandRunner
+}
+
+func newGKECluster() k8sCluster {
+	return &gke{CommandRunner: &defaultRunner{}}
+}
+
+func (g *gke) cloud() string {
+	return "gce"
+}
+
+func (g *gke) getKubeConfig(p *clusterParams) (io.ReadCloser, string, error) {
+	cmd := []string{
+		"gcloud", "container", "clusters", "get-credentials", p.name,
+	}
+	qualifiedClusterName := "gke_"
+	if p.credential != "" {
+		cmd = append(cmd, "--account", p.credential)
+	}
+	if p.project != "" {
+		cmd = append(cmd, "--project", p.project)
+		qualifiedClusterName += p.project + "_"
+	}
+	if p.region != "" {
+		cmd = append(cmd, "--region", p.region)
+		qualifiedClusterName += p.region + "_"
+	}
+	qualifiedClusterName += p.name
+
+	kubeconfig := kubeconfig()
+	result, err := runCommand(g, cmd, kubeconfig)
```

Comment:
thinking if we should write it into a `tmp` kubeconfig, because these might be no need to combine the exported content with existing kubeconfig file.

Reply:
Thought about that (had it originally). But writing to the existing kubeconfig allows the user to then run kubectl against the cluster, which they need to do to dive into errors etc.